modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy

```diff
@@ -0,0 +1,664 @@
+	static private String cacertPath
+	static private WrRestApi client
+
+    @Override
+    void register() {
+        if( session.binDir && !session.binDir.empty() ) {
+            session.abort()
+            throw new AbortOperationException("ERROR: wr executor does not allow the use of custom scripts in the `bin` folder")
+        }
+
+        super.register()
+    }
+
+    protected String getDisplayName() {
+        return "$name [$endpoint]"
+    }
+
+	WrRestApi getClient() {
+		if (!client) {
+			defaultManagerDir = Paths.get(System.getProperty('user.home'), ".wr_production")
+			endpoint = getEndPoint()
+			client = new WrRestApi(endpoint, getToken(), getCacertPath())
+		}
+        client
+    }
+
+    protected String getEndPoint() {
+		// default port that wr listens on is 1021 + (uid * 4) + 1
+		// *** note that this will probably only work on linux/mac os, but wr
+		// probably only works fully on those as well...
+		int uid = ["id", "-u"].execute().text.trim() as Integer
+		int port = 1021 + (uid * 4) + 1
+
+        def result = session.getConfigAttribute('executor.endpoint', "https://localhost:$port")
+        log.debug "[wr] endpoint=$result"
+        return result
+    }
+
+	protected String getToken() {
+        String path = session.getConfigAttribute('executor.tokenpath', Paths.get(defaultManagerDir, "client.token"))
+		log.debug "[wr] tokenpath=$path"
+		String result = new File(path).text
+        return result
+    }
+
+	protected String getCacertPath() {
+        String path = session.getConfigAttribute('executor.cacertpath', Paths.get(defaultManagerDir, "ca.pem"))
+		log.debug "[wr] cacertpath=$path"
+        return path
+    }
+
+    /**
+     * @return {@code false} whenever the containerization is managed by the executor itself
+     */
+    boolean isContainerNative() {
+        return false
+    }
+
+    /**
+     * Create a queue holder for this executor
+     *
+     * @return
+     */
+    TaskMonitor createTaskMonitor() {
+        return WrMonitor.create(session, getClient())
+    }
+
+    /*
+     * Prepare and launch the task in the underlying execution platform
+     */
+    @Override
+    TaskHandler createTaskHandler(TaskRun task) {
+        assert task
+        assert task.workDir
+        log.debug "[wr] Launching process > ${task.name} -- work folder: ${task.workDir}"
+        new WrTaskHandler(task, this)
+    }
+	
+}
+
+
+/**
+ * Handles a job execution using wr as a backend, without needing files on disk
+ * for state.
+ *
+ * @author Sendu Bala <sb10@sanger.ac.uk>
+ * Based on TesTaskHandler by Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ */
+@Slf4j
+@CompileStatic
+class WrTaskHandler extends TaskHandler implements BatchHandler<String,Map> {
+
+    // static final private String WORK_DIR = '/work'
+
+    final List<String> COMPLETE_STATUSES = ['complete', 'buried', 'deleted']
+
+    final List<String> STARTED_STATUSES = ['delayed', 'reserved', 'running'] + COMPLETE_STATUSES
+
```

Comment:
Makes sense to declare these fields also `public static`

Reply:
I made that change, thanks.